In [1]:
import numpy as np

In [2]:
p_hat = 0.85 # sample proportion of success
q_hat = 1 - 0.85 # sample proportion of failure
n = 659

In [9]:
Z_score = 1.96 # Z-score for 95% confidence level
E = Z_score * np.sqrt(p_hat * q_hat / n) # E - margin of error
E

0.0272626743784251

In [6]:
low_bound = p_hat - E
low_bound

0.8227373256215749

In [8]:
upper_bound = p_hat + E
upper_bound

0.8772626743784251

I am 95% sure that p (population prortion of success) is between 0.8227373256215749 and 0.8772626743784251

In [13]:
import statsmodels.api as sm

In [14]:
sm.stats.proportion_confint(n * p_hat, n)

(0.8227378265796143, 0.8772621734203857)

Now, lets take our Cartwheel dataset introduced in lecture and calculate a 95% confidence interval for our mean cartwheel distance:

In [15]:
import pandas as pd
df = pd.read_csv('C:/Users/eli/Desktop/Cartwheeldata.csv')

In [17]:
df.head()

ID  Age Gender  GenderGroup Glasses  GlassesGroup  Height  Wingspan  \
0   1   56      F            1       Y             1    62.0      61.0   
1   2   26      F            1       Y             1    62.0      60.0   
2   3   33      F            1       Y             1    66.0      64.0   
3   4   39      F            1       N             0    64.0      63.0   
4   5   27      M            2       N             0    73.0      75.0   

   CWDistance Complete  CompleteGroup  Score  
0          79        Y              1      7  
1          70        Y              1      8  
2          85        Y              1      7  
3          87        Y              1     10  
4          72        N              0      4

In [18]:
df.describe()

ID        Age  GenderGroup  GlassesGroup     Height   Wingspan  \
count  25.000000  25.000000    25.000000     25.000000  25.000000  25.000000   
mean   13.000000  28.240000     1.520000      0.560000  67.650000  66.260000   
std     7.359801   6.989754     0.509902      0.506623   4.431187   5.492647   
min     1.000000  22.000000     1.000000      0.000000  61.500000  57.500000   
25%     7.000000  24.000000     1.000000      0.000000  64.000000  62.000000   
50%    13.000000  26.000000     2.000000      1.000000  68.000000  66.000000   
75%    19.000000  29.000000     2.000000      1.000000  71.000000  71.000000   
max    25.000000  56.000000     2.000000      1.000000  75.000000  76.000000   

       CWDistance  CompleteGroup      Score  
count   25.000000       25.00000  25.000000  
mean    82.480000        0.76000   6.400000  
std     15.058552        0.43589   2.533114  
min     63.000000        0.00000   2.000000  
25%     70.000000        1.00000   4.000000  
50%     81.000000        1.00000   6.000000  
75%     92.000000        1.00000   8.000000  
max    115.000000        1.00000  10.000000

In [20]:
X_bar = df["CWDistance"].mean() # sample mean for CWDistance
s = df["CWDistance"].std() # sample std for CWDistance
n = len(df)
n

25

In [21]:
Degrees_of_Freedom = n - 1
T_score = 2.064

In [27]:
E = T_score * s / np.sqrt(n) # we don't know population std(σ), we know only sample std(s)
E

6.216170425462932

In [26]:
lower_bound = X_bar - E
lower_bound

76.26382957453707

In [25]:
upper_bound = X_bar + E
upper_bound

88.69617042546294

I am 95% sure that population mean is between 76.26382957453707 and  88.69617042546294 

In [28]:
sm.stats.DescrStatsW(df["CWDistance"]).zconfint_mean()

(76.57715593233026, 88.38284406766975)